tests/unit/development/metadata_test.py

```diff
@@ -90,8 +94,66 @@ class Table(Component):
     _base_nodes = ['optionalNode', 'optionalElement', 'children']
     _namespace = 'TableComponents'
     _type = 'Table'
+    OptionalObjectWithExactAndNestedDescriptionFigure = TypedDict(
+        "OptionalObjectWithExactAndNestedDescriptionFigure",
+            {
+            "data": NotRequired[typing.Union[typing.Sequence[dict], typing.Tuple]],
+            "layout": NotRequired[dict]
+        }
+    )
+
+    OptionalObjectWithExactAndNestedDescription = TypedDict(
+        "OptionalObjectWithExactAndNestedDescription",
+            {
+            "color": NotRequired[str],
+            "fontSize": NotRequired[typing.Union[int, float, numbers.Number]],
+            "figure": NotRequired["OptionalObjectWithExactAndNestedDescriptionFigure"]
+        }
+    )
+
+    OptionalObjectWithShapeAndNestedDescriptionFigure = TypedDict(
+        "OptionalObjectWithShapeAndNestedDescriptionFigure",
+            {
+            "data": NotRequired[typing.Union[typing.Sequence[dict], typing.Tuple]],
+            "layout": NotRequired[dict]
+        }
+    )
+
+    OptionalObjectWithShapeAndNestedDescription = TypedDict(
+        "OptionalObjectWithShapeAndNestedDescription",
+            {
+            "color": NotRequired[str],
+            "fontSize": NotRequired[typing.Union[int, float, numbers.Number]],
+            "figure": NotRequired["OptionalObjectWithShapeAndNestedDescriptionFigure"]
+        }
+    )
+
     @_explicitize_args
-    def __init__(self, children=None, optionalArray=Component.UNDEFINED, optionalBool=Component.UNDEFINED, optionalFunc=Component.UNDEFINED, optionalNumber=Component.UNDEFINED, optionalObject=Component.UNDEFINED, optionalString=Component.UNDEFINED, optionalSymbol=Component.UNDEFINED, optionalNode=Component.UNDEFINED, optionalElement=Component.UNDEFINED, optionalMessage=Component.UNDEFINED, optionalEnum=Component.UNDEFINED, optionalUnion=Component.UNDEFINED, optionalArrayOf=Component.UNDEFINED, optionalObjectOf=Component.UNDEFINED, optionalObjectWithExactAndNestedDescription=Component.UNDEFINED, optionalObjectWithShapeAndNestedDescription=Component.UNDEFINED, optionalAny=Component.UNDEFINED, customProp=Component.UNDEFINED, customArrayProp=Component.UNDEFINED, id=Component.UNDEFINED, **kwargs):
+    def __init__(
+        self,
+        children: typing.Optional[typing.Union[str, int, float, ComponentType, typing.Sequence[typing.Union[str, int, float, ComponentType]]]] = None,
```

Comment:
whoa... :-)

Reply:
We can use [type-aliases](https://docs.python.org/3/library/typing.html#type-aliases) to make our lives easier here. 


```
BasicType: TypeAlias = Union[str, int, float]
ComponentOrBasic: TypeAlias = Union[BasicType, ComponentType]
ChildrenType: TypeAlias = Optional[Union[ComponentOrBasic, Sequence[ComponentOrBasic]]]



children: ChildrenType = None,
````